src/uint/shr.rs

```diff
@@ -139,30 +156,58 @@ mod tests {
 
     #[test]
     fn shr1() {
+        assert_eq!(N.shr1(), N_2);
         assert_eq!(N >> 1, N_2);
     }
 
+    #[test]
+    fn shr256_const() {
+        assert_eq!(N.shr(256), (U256::ZERO, CtChoice::TRUE));
+        assert_eq!(N.shr_vartime(256), (U256::ZERO, CtChoice::TRUE));
+    }
+
+    #[test]
+    #[should_panic(expected = "attempt to shift right with overflow")]
+    fn shr256() {
+        let _ = N >> 256;
+    }
+
     #[test]
     fn shr_wide_1_1_128() {
         assert_eq!(
             Uint::shr_vartime_wide((U128::ONE, U128::ONE), 128),
-            (U128::ONE, U128::ZERO)
+            ((U128::ONE, U128::ZERO), CtChoice::FALSE)
         );
     }
 
     #[test]
     fn shr_wide_0_max_1() {
         assert_eq!(
             Uint::shr_vartime_wide((U128::ZERO, U128::MAX), 1),
-            (U128::ONE << 127, U128::MAX >> 1)
+            ((U128::ONE << 127, U128::MAX >> 1), CtChoice::FALSE)
         );
     }
 
     #[test]
     fn shr_wide_max_max_256() {
         assert_eq!(
             Uint::shr_vartime_wide((U128::MAX, U128::MAX), 256),
-            (U128::ZERO, U128::ZERO)
+            ((U128::ZERO, U128::ZERO), CtChoice::TRUE)
+        );
+    }
+
+    /*
+    #[test]
+    fn shr_limb() {
+        let x = U128::from_be_hex("00112233445566778899aabbccddeeff");
+        assert_eq!(x.shr_limb(0), (x, Limb::ZERO));
+        assert_eq!(
+            x.shr_limb(8),
+            (
+                U128::from_be_hex("0000112233445566778899aabbccddee"),
+                Limb(0xff << (Limb::BITS - 8))
+            )
         );
     }
+    */
```

Comment:
What's up with this?

Reply:
Good catch, commented when I was trying out removing `shr_limb()`. The method is not there anymore, so the test is not necessary.